File: ruby/order/unblock_refund_on_order.rb
require 'ultracart_api'
require_relative '../constants'

#
# unblockRefundOnOrder removes an order property that is considered when a refund request is made.
# If the property is present, the refund is denied.  Being an order property allows for querying
# upon it within BigQuery for audit purposes.#
#
order_api = UltracartClient::OrderApi.new_using_api_key(Constants::API_KEY)

order_id = 'DEMO-0009105222'
order_api.unblock_refund_on_order(order_id)

p 'Method executed successfully.  Returns back 204 No Content'

File: javascript/order/unblockRefundOnOrder.js
import { orderApi } from '../api.js';

/**
 * unblockRefundOnOrder removes an order property that is considered when a refund request is made.
 * If the property is present, the refund is denied.  Being an order property allows for querying
 * upon it within BigQuery for audit purposes.
 */
export class UnblockRefundOnOrder {
    /**
     * Removes a property to prevent an order from being refunded.
     */
    static async execute() {
        const orderId = "DEMO-0009105222";


        try {
            await new Promise((resolve, reject) => {
                orderApi.unblockRefundOnOrder(orderId, function(error, data, response) {
                    if (error) {
                        reject(error);
                    } else {
                        resolve(data);
                    }
                });
            });

            console.log("orderApi.unblockRefundOnOrder executed successfully.");
            // Note: This method returns back a 204 No Content.

        } catch (error) {
            console.error("Error removing block refund on an order.:", error);
        }
    }
}